pkg/object/pipeline/registry.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pipeline
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/megaease/easegress/pkg/context"
+)
+
+type (
+	// Filter is the common interface for filters.
+	Filter interface {
+		// Kind returns the unique kind name to represent itself.
+		Kind() string
+
+		// DefaultSpec returns the default spec.
+		DefaultSpec() interface{}
+
+		// Description returns the description of the filter.
+		Description() string
+
+		// Results returns all possible results, the normal result
```

Comment:
```suggestion
		// Results returns all possible results, excluding the default result value (empty string).
```
Is the suggested comment correct?

Reply:
Yeah, thanks a lot...